File: 62_Unique Paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int row[] = new int[n];
        // init
        for (int i = 0; i < n; i++) row[i] = 1;
        // process
        for (int r = 1; r < m; r++) {
            for (int c = 1; c < n; c++) {
                row[c] += row[c-1];
            }
        }
        return row[n-1];
    }
}

File: 62_Unique Paths.py
class Solution:
    def uniquePaths(self, row: int, col: int) -> int:
        dp_col = [1] * row # update from column to column

        for c in range(1, col):
            for r in range(1, row):
                dp_col[r] += dp_col[r-1]
        return dp_col[-1]